src/theory/strings/rewrites

```diff
@@ -422,6 +495,100 @@
   (str.contains (str.replace s t r) u)
   true)
 
+(define-cond-rule str-replace-dual-ctn-false ((s ?Seq) (t ?Seq) (r ?Seq) (u ?Seq))
+  (and (not (str.contains s t)) (not (str.contains s u)))
+  (str.contains s (str.replace t r u))
+  false)
+
+(define-rule str-replace-self-ctn-simp ((s ?Seq) (t ?Seq))
+  (str.contains s (str.replace t s t))
+  (str.contains s t))
+
+(define-cond-rule str-replace-emp-ctn-src ((s ?Seq) (t ?Seq) (emp ?Seq))
```

Comment:
I don't get this one either. 

Since `emp` is empty:
1. if `s` is empty then `(str.replace emp s t)` is `t` because emp does contain `s` and `t` is preprended.
2. if `s` is non-empty then `(str.replace emp s t)` is empty because emp does not contain s

In the first case the match is false if `t` is empty otherwise true. So that works out.

In the second case, the match is true but the target is false.

Reply:
The target is also true: since `s` is non-empty, then empty replaced by `s` is empty, which is equal to empty.